+++ src/CMakeLists.txt	2023-01-10 11:10:24.304216785 +0100
@@ -255,6 +255,12 @@
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
 endif()
 
+file(COPY include
+    DESTINATION .
+    FILES_MATCHING
+    PATTERN **.h
+)
+
 # ---------------------------------------------------------------------------------------
 # Build binaries
 # ---------------------------------------------------------------------------------------
