+++ src/include/spdlog/common.h	2022-12-13 14:01:59.010744481 +0100
@@ -70,13 +70,7 @@
 #    endif
 #endif
 
-#if defined(__GNUC__) || defined(__clang__)
-#    define SPDLOG_DEPRECATED __attribute__((deprecated))
-#elif defined(_MSC_VER)
-#    define SPDLOG_DEPRECATED __declspec(deprecated)
-#else
 #    define SPDLOG_DEPRECATED
-#endif
 
 // disable thread local on msvc 2013
 #ifndef SPDLOG_NO_TLS
--- src/include/spdlog/fmt/bundled/core.h	2022-12-13 14:00:25.429136614 +0100
+++ src/include/spdlog/fmt/bundled/core.h	2022-12-13 14:02:42.559426824 +0100
@@ -141,17 +141,7 @@
 #endif
 
 #ifndef FMT_DEPRECATED
-#  if FMT_HAS_CPP14_ATTRIBUTE(deprecated) || FMT_MSC_VERSION >= 1900
-#    define FMT_DEPRECATED [[deprecated]]
-#  else
-#    if (defined(__GNUC__) && !defined(__LCC__)) || defined(__clang__)
-#      define FMT_DEPRECATED __attribute__((deprecated))
-#    elif FMT_MSC_VERSION
-#      define FMT_DEPRECATED __declspec(deprecated)
-#    else
 #      define FMT_DEPRECATED /* deprecated */
-#    endif
-#  endif
 #endif
 
 // [[noreturn]] is disabled on MSVC and NVCC because of bogus unreachable code
