+++ src/llvm/cmake/modules/HandleLLVMOptions.cmake	2022-11-11 09:29:00.398786740 +0100
@@ -149,7 +149,11 @@
       set(LLVM_HAVE_LINK_VERSION_SCRIPT 1)
     endif()
   else(FUCHSIA OR UNIX)
-    MESSAGE(SEND_ERROR "Unable to determine platform")
+    if (GENODE)
+      set(LLVM_HAVE_LINK_VERSION_SCRIPT 0)
+    else(GENODE)
+      MESSAGE(SEND_ERROR "Unable to determine platform")
+    endif(GENODE)
   endif(FUCHSIA OR UNIX)
 endif(WIN32)
