+++ src/libcxx/include/CMakeLists.txt	2023-11-02 14:43:52.026814052 +0100
@@ -211,7 +211,7 @@
   set(CXX_HEADER_TARGET cxx-headers)
 endif()
 if(LIBCXX_HEADER_DIR)
-  set(output_dir ${LIBCXX_HEADER_DIR}/include/c++/v1)
+  set(output_dir ${LIBCXX_HEADER_DIR}/include)
 
   set(out_files)
   foreach(f ${files})
@@ -257,7 +257,7 @@
   foreach(file ${files})
     get_filename_component(dir ${file} DIRECTORY)
     install(FILES ${file}
-      DESTINATION ${LIBCXX_INSTALL_HEADER_PREFIX}include/c++/v1/${dir}
+      DESTINATION ${LIBCXX_INSTALL_HEADER_PREFIX}include/${dir}
       COMPONENT ${CXX_HEADER_TARGET}
       PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
     )
@@ -265,7 +265,7 @@
 
   # Install the generated header as __config.
   install(FILES ${LIBCXX_BINARY_DIR}/__generated_config
-    DESTINATION ${LIBCXX_INSTALL_HEADER_PREFIX}include/c++/v1
+    DESTINATION ${LIBCXX_INSTALL_HEADER_PREFIX}include
     PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
     RENAME __config
     COMPONENT ${CXX_HEADER_TARGET})
